syntax = "proto3";

package metatrader;

service OrdersService {

    // Gets an array of the current open orders.
    rpc GetOrders (GetOrdersReq) returns (OrdersList);
}

enum OrderType {
  BUY_MARKET = 0;
  SELL_MARKET = 1;
  BUY_LIMIT = 2;
  SELL_LIMIT = 3;
  BUY_STOP = 4;
  SELL_STOP = 5;
  BUY_STOP_LIMIT = 6;
  SELL_STOP_LIMIT = 7;
  CLOSE_BY = 8;
}

message Order {
    int64 id = 1;
    string symbol = 2;
    OrderType type = 3;
    float price = 4;
    float sl = 5;
    float tp = 6;
    float volumeInitial = 7;
    float volumeCurrent = 8;
    int64 time = 9;
    int64 expiration = 10;
    int64 magic = 11;
    string comment = 12;
}

message GetOrdersReq {
    optional string symbol = 1;
}

message OrdersList {
    repeated Order orders = 1;
}

